streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -178,6 +180,26 @@ public void testProcessOrder() throws Exception {
 
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldAddTimestampExtractorPerSource() throws Exception {
```

Comment:
I found `StreamTaskTest` the best suitable place, as it was suggested to make `SourceNode.getTimestampExtractor()` method available within package. So, it is not accessible inside `TopologyBuilderTest` currently. 
Then I am making `SourceNode.getTimestampExtractor()`  public and moving the tests to `TopologyBuilderTest`.